tests/e2e/idps_test.go

```diff
@@ -29,240 +29,473 @@ var _ = Describe("TF Test", func() {
-			var userName, password string
+		var userName, password,
+			googleIDPClientSecret, googleIDPClientId,
+			gitlabIDPClientSecret, gitlabIDPClientId,
+			githubIDPClientSecret, githubIDPClientId string
 
-			BeforeEach(func() {
+		Describe("IDP Positive scenario test cases", func() {
+			Context("Htpasswd IDP test cases", func() {
 
-				userName = "jacko"
-				password = h.RandStringWithUpper(15)
-				htpasswdMap = []interface{}{map[string]string{"username": userName, "password": password}}
-				idpService.htpasswd = *exe.NewIDPService(con.HtpasswdDir) // init new htpasswd service
+				BeforeEach(func() {
+					userName = "jacko"
+					password = h.RandStringWithUpper(15)
+					htpasswdMap = []interface{}{map[string]string{"username": userName, "password": password}}
+					idpService.htpasswd = *exe.NewIDPService(con.HtpasswdDir) // init new htpasswd service
+				})
+
+				AfterEach(func() {
+					err := idpService.htpasswd.Destroy()
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				Context("Author:smiron-High-OCP-63151 @OCP-63151 @smiron", func() {
+					It("OCP-63151 - Provision HTPASSWD IDP against cluster using TF", ci.Day2, ci.High, ci.FeatureIDP,
+						ci.Exclude,
+						func() {
+							By("Create htpasswd idp for an existing cluster")
+
+							idpParam := &exe.IDPArgs{
+								Token:         token,
+								ClusterID:     clusterID,
+								Name:          "OCP-63151-htpasswd-idp-test",
+								HtpasswdUsers: htpasswdMap,
+							}
+							err := idpService.htpasswd.Create(idpParam)
+							Expect(err).ToNot(HaveOccurred())
+							idpID, _ := idpService.htpasswd.Output()
+
+							By("List existing HtpasswdUsers and compare to the created one")
+							htpasswdUsersList, _ := cms.ListHtpasswdUsers(ci.RHCSConnection, clusterID, idpID.ID)
+							Expect(htpasswdUsersList.Status()).To(Equal(http.StatusOK))
+							respUserName, _ := htpasswdUsersList.Items().Slice()[0].GetUsername()
+							Expect(respUserName).To(Equal(userName))
+
+							By("Login with created htpasswd idp")
+							getResp, err := cms.RetrieveClusterDetail(ci.RHCSConnection, clusterID)
+							Expect(err).ToNot(HaveOccurred())
+							server := getResp.Body().API().URL()
+
+							ocAtter := &openshift.OcAttributes{
+								Server:    server,
+								Username:  userName,
+								Password:  password,
+								ClusterID: clusterID,
+								AdditioanlFlags: []string{
+									"--insecure-skip-tls-verify",
+									fmt.Sprintf("--kubeconfig %s", path.Join(con.RHCS.KubeConfigDir, fmt.Sprintf("%s.%s", clusterID, userName))),
+								},
+								Timeout: 7,
+							}
+							_, err = openshift.OcLogin(*ocAtter)
+							Expect(err).ToNot(HaveOccurred())
+
+						})
+				})
 			})
+			Context("LDAP IDP test cases", func() {
+				BeforeEach(func() {
 
-			AfterEach(func() {
-				err := idpService.htpasswd.Destroy()
-				Expect(err).ToNot(HaveOccurred())
+					userName = "newton"
+					password = "password"
```

Comment:
I wouldn't use a hardcoded password even if it's a negative test case.

Reply:
this specific password must be hardcoded, this is the password for ldap test cases when we need to use 
`"ldap://ldap.forumsys.com/dc=example,dc=com?uid"`

Reply:
(just a suggestion) On the other hand, we can set it as a setting and then use the same code for positive ("newton") and negative test ("notnewton") or something trickier such as a username injection attack.

Reply:
sure, I plan to do several negative tests in the future
I'll insert those aswell